### ISSUE TITLE
Protobuf encoding?

### ISSUE THREAD

**Issue by emiltin** [rsmp-nordic/rsmp_core#202]
Question from a manufacturer:
Did anyone used Google protobufs for validating messages and stronger message typing when using RSMP protocol?

**Comment by emiltin** [comment]
JSON was chosen because it’s very easy to work with and is human readable.

Protocol buffers are more efficient and do provide typing it’s harder to update the format because you need the protocol schema at both ends.

For RSMP 4 we’re considering CBOR, which is a binary encoding of JSON. It’s more efficient, but with the flexibility of JSON, and it does not require a protocol schema.


We do work with JSON schemas to validate the format of messages. Whether it would be easier to work with a protocol schema is worth considering.

**Comment by RazeAntoine** [comment]
When you're communicating in JSON, you may also want to consider whether or not to compress it. Compression can reduce the size of the message by up to 75%. Is it worth looking into?

**Comment by emiltin** [comment]
Yes it's worth considering, however increased CPU need might be problematic on embedded devices.
